person.rb

```diff
@@ -11,12 +14,22 @@ def initialize(age, name = 'unknown', parent_permission: true)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rental = []
   end
 
   def correct_name
     @name
   end
 
+  def classroom=(classroom)
+    @classroom = classroom
+    classroom.add_student(self)
+  end
+
+  def add_new_rental(book, date)
+    @rental << Rental.new(date, book, self)
+  end
```

Comment:
- [Optional] dear @Errytagedesign , great job adding the association between the rental and the person. the only issue I would like to mention is when you create  a new rental you are adding the rental to the person. this functionality ` person.rental << self`. Hence adding the rental to the rental array here is kind of repetitive/. Creating a new rental should be enough to keep the functionalities.

Reply:
Thank you